### DISCUSSION TITLE
Task not awaiting

### DISCUSSION THREAD

**Discussion by ordinaryorange** [Tyrrrz/CliWrap#253] [Q&A]
I cant see what I'm doing wrong here. 
I'm on a windows 11 machine.

```
using CliWrap;
var result = await Cli.Wrap(@"notepad.exe").ExecuteAsync();
Console.WriteLine("Done");
```

Now I know notepad is not a CLI app, but I'm not expecting "Done" to appear before I close notepad ?

**Comment by Tyrrrz** [comment]
GUI apps behave differently on Windows AFAIK and `Process.WaitForExit()` (which CliWrap uses under the hood) does not work the way you would expect.

In general, you shouldn't be using CliWrap with non-CLI applications.

**Reply by ordinaryorange** [reply]
I was using notepad as a test to watch the process open and then for the above code to wait until I closed it.
I've tried this
```
var result = await Cli.Wrap(@"cmd.exe").ExecuteAsync();
Console.WriteLine("Done");
```

But that did not wait either

So I made a .net console app, that does the below

```
Console.WriteLine("Hello, World!");
Console.ReadLine();
```

Then tried 

```
var result = await Cli.Wrap(@"C:\ConsoleApp1\ConsoleApp1\bin\Debug\net8.0\consoleapp1.exe").ExecuteAsync();
Console.WriteLine("Done");
```
But each time "Done" appears in the CliWrap caller before the the processes exits.

**Reply by ordinaryorange** [reply]
Interestingly this works. There must be something special about how cmd and notepad starts.
```
using CliWrap;
var result = await Cli.Wrap(@"charmap.exe").ExecuteAsync();
Console.WriteLine("Done");
```

I tested the above by directly using `Process` class and `WaitForExit` and got the same results, so this odd behaviour is not a CliWrap thing.
